### PR TITLE
Tool to get units associated with any parameter

### PR DESCRIPTION
**Pull request by dlakaplan**
```
import pint.utils
unit = pint.utils.get_unit(p)
```
will return the unit associated with parameter `p`.  Works for normal units, prefix units.

### CODE REVIEW

**Comment by dlakaplan** [comment]
Is this interface OK or would a standalone function work better (say in `pint.utils`)?

**Comment by abhisrkckl** [comment]
I think it will be better in `pint.utils`. I was under the impression that `AllComponents` was not intended for normal use by end users.

**Comment by dlakaplan** [comment]
I don't disagree, but that leads to circular imports

**Comment by abhisrkckl** [comment]
I had the same issue in #1553. I got around that by putting the problematic imports inside the function body.

File: tests/test_parunits.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import pytest
+from astropy import units as u
+from pint.models.timing_model import UnknownParameter
+from pint.utils import get_unit
+
+
+@pytest.mark.parametrize(
+    "p",
+    [
```
[the remark this answers is not in the document]
Comment (dlakaplan) [reply tests/test_parunits.py:9]: DMX was there.  Added the others.

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ and this project, at least loosely, adheres to [Semantic Versioning](https://sem
 - `validate_component_types` method for more rigorous validation of timing model components.
 - roundtrip test to make sure clock corrections are not written to tim files
 - `calc_phase_mean` and `calc_time_mean` methods in `Residuals` class to compute the residual mean.
+- method `AllComponents.param_to_unit` to get units for any parameter
```
[the remark this answers is not in the document]
Comment (dlakaplan) [reply CHANGELOG.md:16]: done

File: tests/test_parunits.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        "XDOT",
+        "EFAC",
+        "EQUAD",
+        "JUMP1",
+    ],
+)
+def test_par_units(p):
+    unit = get_unit(p)
+    print(f"{p}: {unit}")
+    assert isinstance(unit, u.UnitBase) or unit is None
```
Comment (abhisrkckl) [inline tests/test_parunits.py:40]: I think this test can be split into two tests -- one for the case where the output is a unit, and one for the case where it is None.

**Comment by abhisrkckl** [comment]
I think this is good to go. I'll merge it if it's done.

**Comment by dlakaplan** [comment]
Yes, I think it's ready

**Pull request merged**